### PR TITLE
fix(playlists + albums): filters out falsey valued tracks

### PR DESCRIPTION
**Pull request by ColdtQ**
title^ 

Adds a check that makes sure a track isnt null in a album/playlist, if a single track was null, it would cause the whole playlist to not load. This basically ignores the track loading if its null/undefined

### CODE REVIEW

**Comment by ColdtQ** [comment]
The axios commit was from an earlier PR. As of now i've removed the axios and switched it to petitio in the latest change. The first commit can be ignored.

**Comment by Strandxo** [comment]
I'll have a look at this later today.

**Review by anishshobithps — COMMENTED** [review]
I would suggest making a function because its being used in multiple places
```ts
function filterNullOrUndefined(value: unknown): value is null | undefined {
	return value !== undefined || value !== null;
}
```

File: src/plugin.ts [hunk 1]
```diff
@@ -158,12 +158,12 @@ export class Spotify extends Plugin {
 
     private async getAlbumTracks(id: string): Promise<Result> {
         const album = await this.makeRequest<Album>(`${BASE_URL}/albums/${id}`)
-        const tracks = album.tracks.items.map(item => Spotify.convertToUnresolved(item));
+        const tracks = album.tracks.items.filter((item) => item !== undefined || item !== null).map(item => Spotify.convertToUnresolved(item));
```
Comment (anishshobithps) [inline src/plugin.ts]:
```ts
const tracks = album.tracks.items.filter(filterNullOrUndefined).map(item => Spotify.convertToUnresolved(item));
```

File: src/plugin.ts [hunk 2]
```diff
@@ -173,12 +173,16 @@ export class Spotify extends Plugin {
 
     private async getPlaylistTracks(id: string): Promise<Result> {
         const playlist = await this.makeRequest<Playlist>(`${BASE_URL}/playlists/${id}`);
-        const tracks = playlist.tracks.items.map(item => Spotify.convertToUnresolved(item.track));
+        const tracks = playlist.tracks.items.filter((item) =>
```
Comment (anishshobithps) [inline src/plugin.ts]:
```ts
const tracks = playlist.tracks.items.filter(filterNullOrUndefined).map(item => Spotify.convertToUnresolved(item.track));
```

File: src/plugin.ts [hunk 3]
```diff
@@ -173,12 +173,16 @@ export class Spotify extends Plugin {
...
+        item.track !== undefined || item.track !== null
+        ).map(item => Spotify.convertToUnresolved(item.track));
         let next = playlist.tracks.next, page = 1;
 
         while (next && !this.options.playlistLimit ? true : page < this.options.playlistLimit) {
             const nextPage = await this.makeRequest<PlaylistTracks>(next);
-            tracks.push(...nextPage.items.map(item => Spotify.convertToUnresolved(item.track)));
+            tracks.push(...nextPage.items.filter((item) =>
```
Comment (anishshobithps) [inline src/plugin.ts]:
```ts
tracks.push(...nextPage.items.filter(filterNullOrUndefined).map(item => Spotify.convertToUnresolved(item.track)));
```

File: src/plugin.ts [hunk 4]
```diff
@@ -158,12 +158,12 @@ export class Spotify extends Plugin {
...
     private async getAlbumTracks(id: string): Promise<Result> {
         const album = await this.makeRequest<Album>(`${BASE_URL}/albums/${id}`)
-        const tracks = album.tracks.items.map(item => Spotify.convertToUnresolved(item));
+        const tracks = album.tracks.items.filter((item) => item !== undefined || item !== null).map(item => Spotify.convertToUnresolved(item));
         let next = album.tracks.next, page = 1;
 
         while (next && !this.options.playlistLimit ? true : page < this.options.albumLimit) {
             const nextPage = await this.makeRequest<AlbumTracks>(next);
-            tracks.push(...nextPage.items.map(item => Spotify.convertToUnresolved(item)));
+            tracks.push(...nextPage.items.filter((item) => item !== undefined || item !== null).map(item => Spotify.convertToUnresolved(item)));
```
Comment (anishshobithps) [inline src/plugin.ts]:
```ts
tracks.push(...nextPage.items.filter(filterNullOrUndefined).map(item => Spotify.convertToUnresolved(item)));
```

**Review by anishshobithps — CHANGES_REQUESTED** [review]

**Review by ColdtQ — COMMENTED** [review]
I've revised the code to use a function over repeating .filter()

**Comment by ColdtQ** [comment]
Rewrote to use the function @Anish-Shobith suggested

**Pull request merged**